%\VignetteIndexEntry{1. Introduction to MultiDeconfounder}
%\VignetteKeywords{Software, Metagenomics, Microbiome, MultipleComparison, Metabolomics, Transcriptomics}
%\VignettePackage{MultiDeconfoundR}
%\VignetteEngine{knitr::knitr}


\documentclass{article}\usepackage[]{graphicx}\usepackage[usenames,dvipsnames]{color}
% maxwidth is the original width if it is less than linewidth
% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb

\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX
\usepackage{alltt}

\RequirePackage[]{/home/tbirkne/R/x86_64-pc-linux-gnu-library/3.6/BiocStyle/resources/tex/Bioconductor}
\AtBeginDocument{\bibliographystyle{/home/tbirkne/R/x86_64-pc-linux-gnu-library/3.6/BiocStyle/resources/tex/unsrturl}}


%<<setup, echo=FALSE>>=
%suppressPackageStartupMessages({
%    library(BiocParallel)
%    library(BatchJobs)
%    library(VariantAnnotation)
%    library(GenomicAlignments)
%    library(RNAseqData.HNRNPC.bam.chr14)
%    library(TxDb.Hsapiens.UCSC.hg19.knownGene)
%})
%@

\newcommand{\metadeconfoundR}{\Biocpkg{metadeconfoundR}}

\title{Introduction to \metadeconfoundR}
\author{Till Birkner\footnote{\url{tillb@online.de}}, Sofia K. Forslund}
\date{Edited: December 13, 2018; Compiled: \today}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}

\maketitle

\tableofcontents

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\packageVersion{BiocStyle::pkg_ver("metadeconfoundR")}
%\Rpackage{metadeconfoundR} was developed to analyze large scale multi-omics datasets in parallel. It first looks for significant association between omics features like gut microbial OTUs and metadata like drug administration: After sufficient dataset size is checked,  simple association testing is done based on the data type of the respective metadata. %(Fig~\ref{fig:statistics}, left). 
%All three used tests(Wilcoxon/MWU \Rcode{wilcox.test()} for binary, Spearman correlation \Rcode{cor.test()} for continuous numerical, Kruskall \Rcode{kruskal.test()} for neither numerical nor binary) use ranked values. This way assumptions about data distribution are minimized.
%In addition to collecting p-values for all computed tests, effect size is measured if possible. In case of binary data, Cliff's Delta is computed using the \Rcode{orddom::orddom()} function. For continuous data the ``estimate" component from \Rcode{cor.test()} is used.
%
%In a second step, all hits are checked for confounding effects and a status is reported for feature, metadata combination.
 For now please direct questions and bug reports to Till Birkner, a detailed introduction to the package will be added at a later stage.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Quick start}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Install via tar.gz file: \Rcode{install.packages("metadeconfoundR\_0.1.1.tar.gz", repos = NULL)}
%The \Rpackage{MultideconfoundR} package is available at bioconductor.org
%and can be downloaded via \Rcode{BiocManager}:
%<<BiocManager, eval=FALSE>>=
%
%if (!requireNamespace("BiocManager", quietly = TRUE))
%    install.packages("BiocManager")
%BiocManager::install("MultideconfoundR")
%@
%
%Load \MultideconfoundR{}.
%<<MultideconfoundR>>=
%library(MultideconfoundR)
%@
%


%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\section{Use cases}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{\Rcode{sessionInfo()}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{kframe}
\begin{alltt}
\hlkwd{toLatex}\hlstd{(}\hlkwd{sessionInfo}\hlstd{())}
\end{alltt}
\end{kframe}\begin{itemize}\raggedright
  \item R version 3.6.1 (2019-07-05), \verb|x86_64-pc-linux-gnu|
  \item Locale: \verb|LC_CTYPE=en_US.UTF-8|, \verb|LC_NUMERIC=C|, \verb|LC_TIME=de_DE.UTF-8|, \verb|LC_COLLATE=C|, \verb|LC_MONETARY=de_DE.UTF-8|, \verb|LC_MESSAGES=en_US.UTF-8|, \verb|LC_PAPER=de_DE.UTF-8|, \verb|LC_NAME=C|, \verb|LC_ADDRESS=C|, \verb|LC_TELEPHONE=C|, \verb|LC_MEASUREMENT=de_DE.UTF-8|, \verb|LC_IDENTIFICATION=C|
  \item Running under: \verb|Ubuntu 18.04.3 LTS|
  \item Matrix products: default
  \item BLAS:   \verb|/usr/lib/x86_64-linux-gnu/openblas/libblas.so.3|
  \item LAPACK: \verb|/usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so|
  \item Base packages: base, datasets, grDevices, graphics, methods,
    stats, utils
  \item Other packages: metadeconfoundR~0.1.1
  \item Loaded via a namespace (and not attached):
    BiocManager~1.30.8, BiocStyle~2.12.0, R6~2.4.0, Rcpp~1.0.2,
    assertthat~0.2.1, backports~1.1.5, bigmemory~4.5.33,
    bigmemory.sri~0.1.3, callr~3.3.2, cli~1.1.0, codetools~0.2-16,
    commonmark~1.7, compiler~3.6.1, crayon~1.3.4, desc~1.2.0,
    devtools~2.2.1, digest~0.6.21, doParallel~1.0.15,
    ellipsis~0.3.0, evaluate~0.14, foreach~1.4.7, fs~1.3.1,
    glue~1.3.1, grid~3.6.1, htmltools~0.4.0, iterators~1.0.12,
    knitr~1.25, lattice~0.20-38, lmtest~0.9-37, magrittr~1.5,
    memoise~1.1.0, parallel~3.6.1, pkgbuild~1.0.6, pkgload~1.0.2,
    prettyunits~1.0.2, processx~3.4.1, ps~1.3.0, purrr~0.3.2,
    remotes~2.1.0, rlang~0.4.0, rmarkdown~1.16, roxygen2~6.1.1,
    rprojroot~1.3-2, rstudioapi~0.10, sessioninfo~1.1.1,
    stringi~1.4.3, stringr~1.4.0, testthat~2.2.1, tools~3.6.1,
    usethis~1.5.1, withr~2.1.2, xfun~0.10, xml2~1.2.2, yaml~2.2.0,
    zoo~1.8-6
\end{itemize}


\end{document}
